src/Sulu/Bundle/SecurityBundle/Resources/public/js/services/role-router.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+define(['services/husky/mediator'], function(Mediator) {
+
+    'use strict';
+
+    /** @constructor **/
+    function RoleRouter() {}
+
+    RoleRouter.prototype = {
+
+        /**
+         * Navigates to the edit-form.
+         *
+         * @param id The id of the role to edit
+         */
+        toEdit: function(id) {
+            Mediator.emit('sulu.router.navigate', 'settings/roles/edit:' + id + '/details');
+        },
+
+        /**
+         * Navigates to the add-form.
+         */
+        toAdd: function() {
+            Mediator.emit('sulu.router.navigate', 'settings/roles/new');
+        },
+
+        /**
+         * Navigates to the roles list.
+         */
+        toList: function() {
+            Mediator.emit('sulu.router.navigate', 'settings/roles');
+        }
+    };
+
+    return new RoleRouter();
```

Comment:
This could lead to undesired errors, if the component isn't a singleton (if somebody introduces a variable later on)